File: leetcode/C++/daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        stack<int> st;
        vector<int> sol(temperatures.size(),0);
        for(int i=0; i<temperatures.size(); ++i){ 
            while(!st.empty()&&temperatures[st.top()]<temperatures[i]){
                sol[st.top()]=i-st.top();
                st.pop();
            }
            st.push(i);
        }
        return sol;
    }
};

File: leetcode/Python/daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        sol, st = [0 for i in range(len(temperatures))], []
        for i, v in enumerate(temperatures):
            while st and temperatures[st[-1]]<v:
                sol[st.pop()]=i-st[-1]
            st.append(i)
        return sol